### ISSUE TITLE
Classic TRootBrowser: Histogramming leafs which are functions fails

### ISSUE THREAD

**Issue by olifre** [root-project/root#13233]
### Check duplicate issues.

- [X] Checked for duplicates

### Description

Using a ROOT version >= 6.24.06 (older versions above 6.24 may also be affected), double-clicking on leafs e.g. of a `TLorentzVector` stored in a `TTree` (so for example, `X()`) within a classic `TRootBrowser` does not fill the histogram anymore. 

### Reproducer

Create a new file with the following ROOT macro:
```
{
        TFile f("tree.root", "RECREATE");
        TTree tree("NOMINAL", "NOMINAL");
        TLorentzVector my_vector;
        auto branch = tree.Branch("my_vector", &my_vector);
        for (UInt_t i=0; i<100; ++i) {
                my_vector.SetXYZT(i,i,i,i);
                tree.Fill();
        }
        tree.Write();
}
```
Then, enforce the old `TRootBrowser` is used, e.g.:
```
echo "Browser.Name: TRootBrowser" >> ~/.rootrc
```
Finally, open the file inside a `TBrowser`. Then, click on the ROOT file, `NOMINAL`, `my_vector`, and finally double-click on `X()`. 

With the new `ROOT::Experimental::RWebBrowserImp` browser, this works, with the classic `TRootBrowser` in recent ROOT versions, the histogram stays empty, with a manual `Draw()` command, it works. 

### ROOT version

```
   ------------------------------------------------------------------
  | Welcome to ROOT 6.28/04                        https://root.cern |
  | (c) 1995-2022, The ROOT Team; conception: R. Brun, F. Rademakers |
  | Built for linuxx8664gcc on May 08 2023, 02:44:07                 |
  | From tags/v6-28-04@v6-28-04                                      |
  | With                                                             |
  | Try '.help'/'.?', '.demo', '.license', '.credits', '.quit'/'.q'  |
   ------------------------------------------------------------------
```

### Installation method

Gentoo (Portage)

### Operating system

Gentoo Linux

### Additional context

can also be reproduced with 6.24.06 from CVMFS, but not older versions < 6.24.00.

**Comment by bellenot** [comment]
This is because of [this commit](https://github.com/root-project/root/commit/56cfece84e204851d693de8482e2e15c84190bb6). I'm looking at how to re-enable it without re-introducing the issue fixed by this commit...

**Comment by bellenot** [comment]
In fact it's an old issue with `TMethodBrowsable`... Maybe @pcanal could take a look and give some advice how to re-introduce the ability to browse a method, but without recursively adding methods in the browser, as described in the Jira issue [#ROOT-9260](https://sft.its.cern.ch/jira/browse/ROOT-9260)?
 I'm trying to filter which methods are browsable and which are not, but that's tedious...

**Comment by bellenot** [comment]
> Apriori the infinite recursion is "real" as the result of an arbitrary depth of calls "can" be plotted. It feels that the only solution (in addition to filtering the function that are not callable from `Draw` like the operators and function that returns collections) is to delay the populating until the user actually click on the "unfold" button.

But this is already the case...

**Comment by bellenot** [comment]
@pcanal here is the PR: https://github.com/root-project/root/pull/13284

**Comment by bellenot** [comment]
Fixed by commit [e492067bb7](https://github.com/root-project/root/commit/e492067bb7c925612e614473e12e1bfb932e0eb2)
